File: cpp/705_设计哈希集合_easy.cpp
class MyHashSet {
private:
    static const int base = 769;
    vector<list<int>> hashset;
    static int hash(int x) {
        return x % base;
    }
public:
    /** Initialize your data structure here. */
    MyHashSet() : hashset(base) {}
    
    void add(int key) {
        int h = hash(key);
        for (auto it = hashset[h].begin(); it != hashset[h].end(); it++) {
            if ((*it) == key)
                return;
        }
        hashset[h].push_back(key);
    }
    
    void remove(int key) {
        int h = hash(key);
        for (auto it = hashset[h].begin(); it != hashset[h].end(); it++) {
            if ((*it) == key) {
                hashset[h].erase(it);
                return;
            }
        }
    }
    
    /** Returns true if this set contains the specified element */
    bool contains(int key) {
        int h = hash(key);
        for (auto it = hashset[h].begin(); it != hashset[h].end(); it++) {
            if ((*it) == key)
                return true;
        }
        return false;
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */

File: python/705_设计哈希集合_easy.py
class MyHashSet:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.base = 769
        self.data = [[] for i in range(self.base)]
    
    def calHash(self, x):
        return x % self.base

    def add(self, key: int) -> None:
        h = self.calHash(key)
        for i in range(len(self.data[h])):
            if self.data[h][i] == key:
                return
        self.data[h].append(key)

    def remove(self, key: int) -> None:
        h = self.calHash(key)
        for i in range(len(self.data[h])):
            if self.data[h][i] == key:
                self.data[h].remove(key)
                return

    def contains(self, key: int) -> bool:
        """
        Returns true if this set contains the specified element
        """
        h = self.calHash(key)
        for i in range(len(self.data[h])):
            if self.data[h][i] == key:
                return True
        return False


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)